src/Promitor.Agents.Scraper/Scheduling/SchedulingExtensions.cs

```diff
@@ -33,58 +29,86 @@ public static IServiceCollection ScheduleMetricScraping(this IServiceCollection
+                    metricsGroup = new MetricsDeclaration
                     {
-                        ScheduleResourceScraping(resource, metrics.AzureMetadata, metric, azureMonitorClientFactory, metricSinkWriter, runtimeMetricCollector, memoryCache, configuration, azureMonitorLoggingConfiguration, loggerFactory, startupLogger, services);
-                    }
+                        AzureMetadata = allMetrics.AzureMetadata,
+                        MetricDefaults = allMetrics.MetricDefaults
+                    };
+                    metricsGroupings.Add(metric.Scraping, metricsGroup);
                 }
+                
+                metricsGroup.Metrics.Add(metric);
             }
 
-            return services;
+            return metricsGroupings.Values;
         }
 
-        private static void ScheduleResourceScraping(IAzureResourceDefinition resource, AzureMetadata azureMetadata, MetricDefinition metric, AzureMonitorClientFactory azureMonitorClientFactory, MetricSinkWriter metricSinkWriter, IAzureScrapingPrometheusMetricsCollector azureScrapingPrometheusMetricCollector, IMemoryCache memoryCache, IConfiguration configuration, IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory,  ILogger<Startup> logger, IServiceCollection services)
+        private static void ScheduleResourcesScraping(MetricsDeclaration metricsDeclaration,
+            MetricSinkWriter metricSinkWriter,
+            AzureMonitorClientFactory azureMonitorClientFactory,
+            IAzureScrapingPrometheusMetricsCollector azureScrapingPrometheusMetricsCollector,
+            IMemoryCache resourceMetricDefinitionMemoryCache,
+            IScrapingMutex scrapingTaskMutex,
+            IConfiguration configuration,
+            IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration,
+            ILoggerFactory loggerFactory,
+            ILogger<Startup> logger,
+            IServiceCollection services)
         {
-            var resourceSubscriptionId = string.IsNullOrWhiteSpace(resource.SubscriptionId) ? azureMetadata.SubscriptionId : resource.SubscriptionId;
-            var azureMonitorClient = azureMonitorClientFactory.CreateIfNotExists(azureMetadata.Cloud, azureMetadata.TenantId, resourceSubscriptionId, metricSinkWriter, azureScrapingPrometheusMetricCollector, memoryCache, configuration, azureMonitorLoggingConfiguration, loggerFactory);
-            var scrapeDefinition = metric.CreateScrapeDefinition(resource, azureMetadata);
-            var jobName = GenerateResourceScrapingJobName(scrapeDefinition, resource);
-
+            var jobName = GenerateResourceScrapingJobName(metricsDeclaration, logger);
+            
             services.AddScheduler(builder =>
             {
                 builder.AddJob(jobServices =>
-                    {
-                        return new ResourceScrapingJob(jobName, scrapeDefinition,
+                        new ResourcesScrapingJob(jobName,
```

Comment:
Why do we need all of these additional dependencies?

Reply:
They were part of your original implementation, I just mirrored the code needed previously for ResourceDiscoveryGroupScrapingJob. Do you think something more should be changed?